lib/node/failure_detector.ml

```diff
@@ -63,17 +63,24 @@ let rec pick_random_neighbors neighbors number_of_neighbors =
       elem :: pick_random_neighbors neighbors (number_of_neighbors - 1)
 
 (** Updates a peer in the node's peer list with
-the given status. Returns a result that contains
-unit if the peer is found in the node's list, and
-a string stating that a peer with the given address
-could not be found otherwise. *)
+   the given status. Returns a result that contains
+   unit if the peer is found in the node's list, and
+   a string stating that a peer with the given address
+   could not be found otherwise. *)
 let update_peer_status node peer status =
   let open Peer in
   let neighbor = Base.Hashtbl.find !node.peers peer.address in
   match neighbor with
-  | Some neighbor ->
+  | Some neighbor -> (
```

Comment:
How about

```ocaml
match neighbor with
| Some neighbor when status = Suspicious ->
  let time = Unix.gmtime @@ Unix.time () in
  Result.Ok(neighbor.last_suspicious_status <- Some time)
| Some neighbor ->
  Result.Ok(neighbor.last_suspicious_status <- None)
| None ->
  Result.Error (Printf.sprintf "Failed to find peer with address %s:%d in node peer list"
         peer.address.address peer.address.port)
```

Reply:
Yup! Didn't know this syntax, looks great.
Done.